src/AvtaleSide/MaalSteg/RedigerMaal/RedigerMaal.tsx

```diff
@@ -30,6 +35,60 @@ class RedigerMaal extends React.Component<Props, State> {
             '',
         beskrivelseFeil: undefined,
         valgtKategoriFeil: undefined,
+        erLagret: false,
+    };
+
+    componentDidMount(): void {
+        if (this.props.mellomLagretData) {
+            if (this.props.mellomLagretData.maalTekst !== '') {
+                this.setState({
+                    valgtKategori: this.mapKategoriTilMaal(
+                        this.props.mellomLagretData.maal
+                    ),
+                    beskrivelse: this.props.mellomLagretData.maalTekst,
+                });
+            }
+        }
+    }
+
+    componentWillUnmount(): void {
+        const liste = this.props.ledigeMaalkategorier
+            .filter(mal => mal !== this.state.valgtKategori)
+            .map(mal => mal);
+        if (this.state.beskrivelse !== '' && !this.state.erLagret) {
+            const tempMaal = {
+                maal: this.state.valgtKategori
+                    ? this.state.valgtKategori
+                    : liste[0],
+                maalTekst: this.state.beskrivelse,
+            };
+            if (this.props.setMellomLagring) {
+                this.props.setMellomLagring(tempMaal);
+            }
+        } else if (!this.state.valgtKategori && this.state.beskrivelse === '') {
+            if (this.props.fjernMellomLagring) {
+                this.props.fjernMellomLagring();
+            }
+        }
+    }
+
+    mapKategoriTilMaal = (input: string): Maalkategori => {
+        switch (input) {
```

Comment:
Går det an å typesette `input` som `Maalkategori` for å slippe denne metoden?

Reply:
vel ikke gunstig å typesette input som maalkategori. Sette default som tom string ved initialisering og ved lagring som den sjekkes på. Definerer du input som maalKategori så må en bruke ekstra boolean i Context Api som sjekk for å vite om det er data som er mellomlagret eller ikke.